lib/srv/server/azure_installer.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/gravitational/trace"
+	"golang.org/x/sync/errgroup"
+
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/cloud/azure"
+)
+
+// AzureInstallerConfig represents configuration for an Azure install script
+// executor.
+type AzureInstallerConfig struct {
+	Emitter     apievents.Emitter
+	AccessPoint auth.DiscoveryAccessPoint
+}
+
+// AzureInstaller handles running commands that install Teleport on Azure
+// virtual machines.
+type AzureInstaller struct {
+	AzureInstallerConfig
+}
+
+// NewAzureInstaller returns a new Azure installer.
+func NewAzureInstaller(cfg AzureInstallerConfig) *AzureInstaller {
+	return &AzureInstaller{
+		AzureInstallerConfig: cfg,
+	}
+}
+
+// AzureRunRequest combines parameters for running commands on a set of Azure
+// virtual machines.
+type AzureRunRequest struct {
+	Client          azure.RunCommandClient
+	Instances       []*armcompute.VirtualMachine
+	Params          []string
+	Region          string
+	ResourceGroup   string
+	ScriptName      string
+	PublicProxyAddr string
+}
+
+// Run runs a command on a set of virtual machines and then blocks until the
+// commands have completed.
+func (ai *AzureInstaller) Run(ctx context.Context, req AzureRunRequest) error {
```

Comment:
No test coverage?

Reply:
I'm not sure how we would. This is basically a wrapper for `azure.RunCommandClient.Run`, which also isn't very amenable to unit testing.